[new product] Add Apache NiFi (#6082)
captn3m0/endoflife.date: products/apache-nifi.md

---
title: Apache NiFi
category: server-app
tags: apache java-runtime
iconSlug: apachenifi
permalink: /apache-nifi
alternate_urls:
-   /nifi
changelogTemplate: "https://cwiki.apache.org/confluence/display/NIFI/Release+Notes#ReleaseNotes-Version__LATEST__"
eolColumn: Support
releaseDateColumn: true

identifiers:
-   repology: nifi

auto:
  methods:

>   -   git: https://github.com/apache/nifi.git

Reviewer: ```suggestion
  -   git: https://github.com/apache/nifi.git
      regex: '^rel\/nifi-(?P<major>\d+)\.(?P<minor>\d+)\.(?P<patch>\d+)$'
```

The pattern used on https://github.com/apache/nifi/tags does not match the default pattern.

releases:

> -   releaseCycle: "1.27.0"

Reviewer: ```suggestion
-   releaseCycle: "1.27"
```

If Apache Nifi follows semver, 1.27 is the release. This comment applies to all releases.

    releaseDate: 2024-07-08
    eol: false
    latest: "1.27.0"

> -   releaseCycle: "2.0.0-M4"
>     releaseDate: 2024-07-01
>     eol: false
>     latest: "2.0.0-M4"
> -   releaseCycle: "2.0.0-M3"
>     releaseDate: 2024-05-16
>     eol: 2024-07-02
>     latest: "2.0.0-M3"

Reviewer: ```suggestion
```

Beta, RC, milestone... are never documented on endoflife.date.

-   releaseCycle: "1.26.0"
    releaseDate: 2024-05-06
    eol: 2024-07-09

>     latest: "1.26.0"

Reviewer: ```suggestion
    latest: "1.26.0"
    latestReleaseDate: 2024-05-06
```

All releases are missing the `latestReleaseDate` field (can be found on https://github.com/apache/nifi/tags).

-   releaseCycle: "1.25.0"
    releaseDate: 2024-01-29
    eol: 2024-05-07
    latest: "1.25.0"
-   releaseCycle: "1.24.0"
    releaseDate: 2023-11-27
    eol: 2024-01-30
    latest: "1.24.0"

> -   releaseCycle: "2.0.0-M2"
>     releaseDate: 2023-11-25
>     eol: 2024-05-17
>     latest: "2.0.0-M2"
> -   releaseCycle: "2.0.0-M1"
>     releaseDate: 2023-11-25
>     eol: 2024-01-30
>     latest: "2.0.0-M1"

Reviewer: ```suggestion
```

Beta, RC, milestone... are never documented on endoflife.date.

-   releaseCycle: "1.23.0"
    releaseDate: 2023-07-28
    eol: 2023-11-28
    latest: "1.23.2"
    latestReleaseDate: 2023-08-22
-   releaseCycle: "1.22.0"
    releaseDate: 2023-06-11
    eol: 2023-07-29
    latest: "1.22.0"
-   releaseCycle: "1.21.0"
    releaseDate: 2023-04-07
    eol: 2023-06-12
    latest: "1.21.0"
-   releaseCycle: "1.20.0"
    releaseDate: 2023-02-09
    eol: 2023-04-08
    latest: "1.20.0"

> -   releaseCycle: "1.19.0"

Reviewer: A bank line between releases would improve readability.

    releaseDate: 2022-11-28
    eol: 2023-02-10
    latest: "1.19.1"
    latestReleaseDate: 2022-12-07

---

> > [Apache NiFi](https://nifi.apache.org/) is an easy to use, powerful, and reliable system 
> > to process and distribute data.

Reviewer: ```suggestion
> [Apache NiFi](https://nifi.apache.org/) is a software project designed to automate the
> flow of data between software systems. 
```

I find the description [from Wikipedia](https://en.wikipedia.org/wiki/Apache_NiFi) more neutral.

> Apache NiFi has no published information on supported releases. This page marks the latest minor release as supported
> and other releases as EOL on the day of the succeeding release.

Reviewer: ```suggestion
The support and EOL policy is not documented. Looking at past releases, it seems that only the latest release is actively maintained.
```

Not a big fan of the `This page marks the latest minor release as supported...`. Is it true to say this instead ? This was taken from https://endoflife.date/jquery-ui, which similarly have no support policy.

> The full list of versions, with their associated release notes, is available [on the Apache NiFi
> Wiki](https://cwiki.apache.org/confluence/display/NIFI/Release+Notes).

Reviewer: ```suggestion
```

Each release will have it's link, so I don't think this paragraph is needed. WDYT ?
